tybalt/models.py

```diff
@@ -324,19 +342,25 @@ def _compile_adage(self):
 
     def _connect_layers(self):
         # Separate out the encoder and decoder model
-        self.encoder = Model(self.input_rnaseq, self.encoded)
-
         encoded_input = Input(shape=(self.latent_dim, ))
         decoder_layer = self.full_model.layers[-1]
         self.decoder = Model(encoded_input, decoder_layer(encoded_input))
 
+        if self.tied_weights:
+            self.encoder = Model(self.encoded.input, self.encoded.output)
+        else:
+            self.encoder = Model(self.input_rnaseq, self.encoded)
```

Comment:
Clarification Q: what is self.encoded.input?

Reply:
There are two ways to build a keras graph:

1. Adding layer by layer with a `Sequential` model
2. Connecting layers directly

The tied weights model uses 1 while the untied model uses 2. So, `self.encoded.input` and `self.encoded.output` are the two Tensors that fall out of the `self.encoded` layer. A `Model` is connected by `Tensors`. I added c4b7c82 to clarify.